src/main/jbake/content/blog/2017/08/opentracing-appmetrics-canary.adoc

```diff
@@ -0,0 +1,157 @@
+```
+
+This result of this query can be seen in the image at the top of the article. This chart shows that version `0.0.2` of the
+`accountmgr` service has not generated any errors, while the `0.0.2` of the `ordermgr` appears to be less error prone that
+version `0.0.1`.
+
+Based on these results we may conclude that these versions are better, or atleast no worse, than the previous version - and
+so we should update the main service deployments to use the new version. In the OpenShift UI this can be achieved by
+selecting the three vertical dots on the right hand side of the deployment region, and select the "Edit Yaml" menu item. This
+will display an editor window where the version can be changed from `0.0.1` to `0.0.2`.
+
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-08-18-canary-service-update.png[caption="Figure 2: ", title="Update the service version"]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-08-18-canary-service-update.png[caption="Figure 2: ", title="Update the service version"]
+endif::[]
+
+Once this is saved, the OpenShift UI will show that the service is going through a "rolling update", to incrementally change
+each service instance over to the new version.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-08-18-canary-rolling-update.png[caption="Figure 3: ", title="Rolling update"]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-08-18-canary-rolling-update.png[caption="Figure 3: ", title="Rolling update"]
+endif::[]
+
+Once this has been done for both the `ordermgr` and `accountmgr` services, then the canary version of each deployment
+can be scaled down or removed completely.
+
+An alternative to performing the rolling update would simply be to name the canary version something else (i.e. specific
+to the version being tested), and when it comes time to switch over, simply scale down the prevous deployment version. This
+would be more straightforward, but wouldn't show off the cool rolling update approach in the OpenShift UI :)
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-08-18-canary-scale-down.png[caption="Figure 4: ", title="Scaling down canary deployment"]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-08-18-canary-scale-down.png[caption="Figure 4: ", title="Scaling down canary deployment"]
+endif::[]
+
+NOTE: Although we have updated both services at the same time, this is not necessary. Generally microservices would be managed
+by separate teams and subject to their own deployment lifecycles.
+
+
+== Conclusion
+
+This article has shown how application metrics, captured by instrumenting services using the OpenTracing API, can
+be used to support a simple Canary deployment strategy.
```

Comment:
> strategy, where we assess the quality of new versions on only a portion of the live traffic.

Perhaps add a link to the "official" explanation about Canary Releases? https://martinfowler.com/bliki/CanaryRelease.html